src/hooks/useGoodsItemListQuery.ts

```diff
@@ -0,0 +1,31 @@
+import { getThemesProducts } from '@apis/themes';
+import { ThemeProductsRequest } from '@/types/requestTypes';
+import { ThemeProductsResponse } from '@/types/responseTypes';
+import { QueryFunctionContext, useInfiniteQuery } from '@tanstack/react-query';
+import { AxiosError } from 'axios';
+import { useMemo } from 'react';
+
+interface useGoodsItemListQueryProps {
+  themeKey: string;
+  rowsPerPage: number;
+}
+
+export default function useGoodsItemListQuery({ themeKey, rowsPerPage }: useGoodsItemListQueryProps) {
+  const queryKey = ['theneProduct', themeKey];
+  const queryFn = ({ pageParam = '0' }: QueryFunctionContext) =>
+    getThemesProducts({ themeKey, pageToken: pageParam, maxResults: rowsPerPage } as ThemeProductsRequest);
+
+  const { data, isLoading, isError, error, fetchNextPage, isFetchingNextPage, hasNextPage } = useInfiniteQuery<
+    ThemeProductsResponse,
+    AxiosError
+  >({
+    queryKey,
+    queryFn,
+    initialPageParam: '0',
```

Comment:
```suggestion
    queryKey: ['themeProduct', themeKey],
    queryFn: ({ pageParam = '0' }) => {
      return getThemesProducts({ themeKey, pageToken: pageParam, maxResults: rowsPerPage });
    },
```

queryKey, queryFn은 object value로 바로 전달하지 않고 변수 선언해서 따로 분리해서 전달하게 되면 QueryFunctionContext처럼 타입 직접 지정해줘야해서 번거로울 수 있어요. 이런 경우에는 위에 제안드린 코드처럼 직접 전달하는걸 추천해요.

지금보니까 `useInfiniteQuery<ThemeProductsResponse, AxiosError>`에 전달한 타입이 잘못됐는지 pageParam이 `unknown`으로 뜨네요. 한번 개선해보시겠어요? 제안드린 코드 적용하고 `useInfiniteQuery`의 타입 변수에 아무런 타입 전달하지 않으니 잘 추론되는거 같네요.